BigBaseV2/src/hooks/protections/received_clone_sync.cpp

```diff
@@ -20,21 +35,317 @@ namespace big
+	{
+		auto const fnv_offset_basis = 14695981039346656037ULL;
+		auto const fnv_prime = 1099511628211ULL;
+
+		auto value = fnv_offset_basis;
+		for (auto i = 0; i < len; i++) {
+			value ^= static_cast<size_t>(str[i]);
+			value *= fnv_prime;
+		}
+		value ^= value >> 32;
+
+		return value;
+	}
+
+	uint32_t crash_models[] = { RAGE_JOAAT("prop_dummy_01"), RAGE_JOAAT("prop_dummy_car"), RAGE_JOAAT("prop_dummy_light"), RAGE_JOAAT("prop_dummy_plane"), RAGE_JOAAT("slod_human"),
+		RAGE_JOAAT("slod_small_quadped"), RAGE_JOAAT("slod_large_quadped"), RAGE_JOAAT("prop_distantcar_night"), RAGE_JOAAT("prop_distantcar_day"), RAGE_JOAAT("hei_bh1_08_details4_em_night"),
+		RAGE_JOAAT("dt1_18_sq_night_slod"), RAGE_JOAAT("ss1_12_night_slod"), -1288391198, RAGE_JOAAT("h4_prop_bush_bgnvla_med_01"), RAGE_JOAAT("h4_prop_bush_bgnvla_lrg_01"),
+		RAGE_JOAAT("h4_prop_bush_buddleia_low_01"), RAGE_JOAAT("h4_prop_bush_ear_aa"), RAGE_JOAAT("h4_prop_bush_ear_ab"), RAGE_JOAAT("h4_prop_bush_fern_low_01"),
+		RAGE_JOAAT("h4_prop_bush_fern_tall_cc"), RAGE_JOAAT("h4_prop_bush_mang_ad"), RAGE_JOAAT("h4_prop_bush_mang_low_aa"), RAGE_JOAAT("h4_prop_bush_mang_low_ab"),
+		RAGE_JOAAT("h4_prop_bush_seagrape_low_01"), RAGE_JOAAT("prop_h4_ground_cover"), RAGE_JOAAT("h4_prop_weed_groundcover_01"), RAGE_JOAAT("h4_prop_grass_med_01"),
+		RAGE_JOAAT("h4_prop_grass_tropical_lush_01"), RAGE_JOAAT("h4_prop_grass_wiregrass_01"), RAGE_JOAAT("h4_prop_weed_01_plant"), RAGE_JOAAT("h4_prop_weed_01_row"),
+		RAGE_JOAAT("urbanweeds02_l1"), RAGE_JOAAT("proc_forest_grass01"), RAGE_JOAAT("prop_small_bushyba"), RAGE_JOAAT("arbitergt"), RAGE_JOAAT("astron2"), RAGE_JOAAT("cyclone2"),
+		RAGE_JOAAT("ignus2"), RAGE_JOAAT("s95"), RAGE_JOAAT("hc_gunman"), RAGE_JOAAT("v_res_d_dildo_a"), RAGE_JOAAT("v_res_d_dildo_b"), RAGE_JOAAT("v_res_d_dildo_c"),
+		RAGE_JOAAT("v_res_d_dildo_d"), RAGE_JOAAT("v_res_d_dildo_e"), RAGE_JOAAT("v_res_d_dildo_f"), RAGE_JOAAT("v_res_skateboard"), RAGE_JOAAT("prop_battery_01"), RAGE_JOAAT("prop_barbell_01"),
+		RAGE_JOAAT("prop_barbell_02"), RAGE_JOAAT("prop_bandsaw_01"), RAGE_JOAAT("prop_bbq_3"), RAGE_JOAAT("v_med_curtainsnewcloth2"), RAGE_JOAAT("bh1_07_flagpoles"),
+		92962485, RAGE_JOAAT("ig_wade") };
+
+	uint32_t cage_models[] = { RAGE_JOAAT("prop_rub_cage01a"), RAGE_JOAAT("prop_fnclink_05crnr1"), RAGE_JOAAT("prop_gold_cont_01"), RAGE_JOAAT("prop_feeder1"), RAGE_JOAAT("stt_prop_stunt_tube_s"),
+		RAGE_JOAAT("prop_feeder1_cr"), RAGE_JOAAT("p_cablecar_s") };
+
+	inline bool is_model_a_crash_model(uint32_t model) {
+		for (int i = 0; i < sizeof(crash_models) / sizeof(uint32_t); i++)
+		{
+			if (crash_models[i] == model) return true;
+		}
+		return false;
+	}
+
+	inline bool is_model_a_cage_model(uint32_t model) {
+		for (int i = 0; i < sizeof(cage_models) / sizeof(uint32_t); i++)
+		{
+			if (cage_models[i] == model) return true;
+		}
+		return false;
+	}
 	
+	CPhysicalAttachDataNode* get_attach_node_from_object(rage::netSyncNodeBase* node)
```

Comment:
Possibly offer a function that replaces `get_attach_node_from_object` and `get_ped_attach_node_from_object`, though I'm not familiar with these protections so it might not be possible to implement such a function.
```cpp
template<typename T>
T* get_data_node(rage::netSyncNodeBase* node, uint64_t hash_code)
{
	if (node->IsParentNode())
	{
		for (auto child = node->m_first_child; child; child = child->m_next_sibling)
		{
			T* data_node = get_data_node<T>(child);
			if (data_node != nullptr)
				return data_node ;
		}
	}
	else if (node->IsDataNode())
	{
		if (typeid(*node).hash_code() == hash_code) //CPedAttachDataNode
			return reinterpret_cast<T*>(node);
	}
	return nullptr;
}
```

usage
```
const auto ped_attach_data_node = get_data_node<CPedAttachDataNode>(node, "?AVCPedAttachDataNode@@"_fnv1a);
```